src/tutorials/0006-anatomy-of-a-cid/06.md

```diff
@@ -0,0 +1,56 @@
+
+Looking at the results from the CID Inspector tool we can see several parts that the tool was able to parse for us:
+
+- `Human Readable CID`: breaks down each part of the CID to be easily readable by us humans
+- `Multibase`: `code` is the identifier of the base, in this case `b` for `base32`.
+- `Multicodec`: `code` is the identifier of the codec, in this case `0x70` for `dag-pb`, an IPLD format
+- `Multihash`: breakdown of the multihash into the hashing algorithm used (`18` is the code for `sha2-256`) and the length of the hash in bits (`256`)
+
+From the "Human Readable CID" breakdown, we can see that the original hash of the content, before the appropriate CIDv1 prefixes are added, is `c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a`.
+
+## Example 2: CIDv0
+
+[`QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`](https://cid.ipfs.io/#QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR)
+
+![Results from the CID Inspector tool for the first example](tutorial-assets/T0006L06-example-2.jpg)
+
+This Version 0 CID shows some different results: both the `multibase` and the `multicodec` are listed as "implicit".
+Since Version 0 CIDs did not have those prefixes, they are always assumed to be `base58btc` and `dag-pb` respectively.
+
+You may have noticed that the `Base32 CIDV1` part in the end is `bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi`, which is the same CID from the first example!
+
+Notice also how the end of the "Human Readable CID" (the portion after the prefixes) is exactly the same in this CIDv0 example as it was in the CIDv1 example: `c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a`.
+
+Why? These two CIDs point to the same content. Basically, it's the same hash (`c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a`) represented in the two different versions of the CID spec.
+
+You can convert any `CIDv0` to `CIDv1`, because the implicit prefixes from `v0` become explicit in `v1`.
+However, because `CIDv1` supports multiple codecs and multiple bases and `CIDv0` does not, not all `CIDv1` can be converted to `CIDv0`. In fact, only `CIDv1` that have the following properties can be converted to `CIDv0`:
+
+- `multibase = base58btc`
+- `multicodec = dag-pb`
+- `multihash-algorithm = sha2-256`
+- `multihash-length = 256`
+
+To test this theory, you can check out our beloved `Aardvark` page here, hosted on the IPFS network: https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Aardvark.html
+
+- Open the link in your browser and copy the CID from the middle of the URL (`QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco`)
+- In a new browser window, paste it into the [CID Inspector tool](https://cid.ipfs.io) and find the equivalent CIDv1 value displayed at the bottom of the screen
+- Back in your aardvark tab, replace the `v0` CID with the converted `v1` CID in the original URL and refresh the page
+
+You should see the same article on our aardvark.
```

Comment:
I'd like to see a summary here explaining the final anatomy of a CID with an illustration. Something like this:

```markdown
Summary

So when presented with a CID, you should now be able to find the following information:

1. The hash of the content itself.
2. The length of that hash.
3. The hashing function used to hash the content.
4. The encoding method used to read the content.
5. The CID version this CID is built from.
6. And if the CID is written as a string (not as binary), what type of base encoding was applied to the binary data.

![Example of a CID with each section listed and coloured.]()
```

The image could be something like this hastily draw picture courtesy of Microsoft paint:

![Untitled](https://user-images.githubusercontent.com/9611008/75377947-c9b02f80-58a0-11ea-8b0c-37140ccd5c02.png)

This image is 100% not accurate. I just threw it together to illustrate what I think would work best here.

Reply:
@johnnymatthews sadly this is not possible: there is no direct character-for-character correlation between a base32 encoding and the underlying bytes.

Perhaps we could set up a 15 min demo with cyberchef to give you folks some intuition on how the pieces build on top of each other.

Reply:
Ah, that's a shame. Would the above be possible for a binary implementation?

Reply:
Yep! ( if by `implementation` you meant `representation` )

Basically for what you are asking to work, the requirement is that **the source base** ( in our case `256`: 8 bits in a byte == `2^8` == `256` ) **can be reduced by using only the √** operation to **the target base**. Therefore in your case all you have is`16`, `4`, and `2`: i.e. hex( `\x01` `\xE2` ... ) , nibbles ( `\x1` `\xA` `\xF` ... ) and binary ( `1` `1` `0` `1` ... ). Any other base ( 32, 58, etc ) will be "misaligned" by a certain cyclic amount of characters.

This is also why you often hear that our "native" base of 10 doesn't fit very well into computers ( √10 is not an integer )

Reply:
@johnnymatthews another way to look at it for bytes vs base32 encoding: https://upload.wikimedia.org/wikipedia/commons/f/fe/Base32_DE.svg

Reply:
@johnnymatthews Understanding that we couldn't do it to a picture of the string CID and would have to do it to a picture of the binary, would you still find this summary diagram helpful? I'm assume it would be some combination of these three formats of diagram, but extended out to include every type of prefix except base prefix, since that only exists in strings: 
![image](https://user-images.githubusercontent.com/19171465/75720730-04e99e80-5ca5-11ea-90a9-e9250d07a74e.png)

Note that we already have 
![image](https://user-images.githubusercontent.com/19171465/75720893-4d08c100-5ca5-11ea-854c-53dd46c0ab2b.png)
and 
![image](https://user-images.githubusercontent.com/19171465/75720907-54c86580-5ca5-11ea-987a-95d79b5c8d46.png)
as our closest-to-complete diagrams in other formats throughout the lessons.